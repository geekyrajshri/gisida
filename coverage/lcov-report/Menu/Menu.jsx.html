<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Menu/Menu.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">Menu</a> Menu.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">71.43% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/6</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">require('./Menu.scss');
&nbsp;
import React from 'react';
import Cookie from 'js-cookie';<span class="branch-0 cbranch-no" title="branch not covered" ></span>
&nbsp;
const logOut = (e) =&gt; <span class="fstat-no" title="function not covered" >{</span>
<span class="cstat-no" title="statement not covered" >  e.preventDefault();</span>
  Cookie.set('dsauth', false);
<span class="cstat-no" title="statement not covered" >  location.reload();</span>
};
&nbsp;
const Menu = ({ toggleSingleScreen = f =&gt; f, toggleSplitScreen = f =&gt; f, appConfig = f =&gt;f }) =&gt;
  (&lt;div className="menu"&gt;
    &lt;div className="brand"&gt;
      &lt;img src={appConfig.appIcon} alt="UKAID" className="brand-icon" /&gt;
      &lt;div className="brand-title"&gt;
        &lt;span className="white"&gt;{appConfig.appName}&lt;/span&gt;&amp;nbsp;&amp;nbsp; {appConfig.appNameDesc}&lt;/div&gt;
    &lt;/div&gt;
    &lt;ul&gt;
      &lt;li&gt;&lt;a href="#" onClick={toggleSingleScreen}&gt;&lt;span className="glyphicon glyphicon-home" /&gt;&lt;/a&gt;&lt;/li&gt;
      &lt;li&gt;&lt;a href="#" onClick={toggleSplitScreen}&gt;&lt;span className="glyphicon glyphicon-th-large" /&gt;&lt;/a&gt;&lt;/li&gt;
    &lt;/ul&gt;
    &lt;a className="sign-out" onClick={logOut}&gt;&lt;span className="glyphicon glyphicon-log-out" /&gt;&lt;/a&gt;
  &lt;/div&gt;);
&nbsp;
export default Menu;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jun 27 2017 15:27:56 GMT+0300 (EAT)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
