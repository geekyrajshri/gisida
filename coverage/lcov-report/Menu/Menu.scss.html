<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Menu/Menu.scss</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">Menu</a> Menu.scss
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/0</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">.menu {
  width: 80px;
  background: #E74624;
  height: 100%;
  position: fixed;
  top: 0;
  left: 0px;
  z-index: 1000;
  box-shadow: 0 1px 7px rgba(0, 0, 0, 0.4);
  
  ul {
    padding: 0;
    list-style-type: none;
    
    li a {
      display: block;
      width: 80px;
      line-height: 3.5;
      color: #fff;
      font-size: 20px;
      text-align: center;
    }
  }
}
&nbsp;
.sign-out {
  position: absolute;
  bottom: 30px;
  left: 0;
  font-size: 18px;
  width: 80px;
  padding: 15px 0;
  display: block;
  color: #fff;
  font-weight: bold;
  text-align: center;
  cursor: pointer;
}
.sign-out:hover,
.sign-out:focus {
  color: #999;
}
&nbsp;
&nbsp;
.brand-title {
  color: #fff;
  font-size: 14px;
  position: absolute;
  bottom: 100px;
  left: 28px;
  display: block;
  writing-mode: vertical-rl;
  -webkit-writing-mode: vertical-rl;
}
.brand-icon {
  display: block;
  width: 80px;
  height: 70px;
  text-align: center;
  color: #ccc;
  margin-top: 10px;
  background: no-repeat center center;
  background-size: 70%;
  text-indent: -9999px;
}
.white {
  text-transform: uppercase;
}</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jun 27 2017 15:27:56 GMT+0300 (EAT)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
