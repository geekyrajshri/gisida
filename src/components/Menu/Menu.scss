.menu {
  width: 80px;
  background: #e74624;
  height: 100%;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 1000;
  box-shadow: 0 1px 7px rgba(0, 0, 0, 0.4);

  ul {
    padding: 0;
    list-style-type: none;

    li a {
      display: block;
      width: 80px;
      line-height: 3.5;
      color: #fff;
      font-size: 20px;
      text-align: center;
    }
  }
}

.sign-out {
  position: absolute;
  bottom: 30px;
  left: 0;
  font-size: 18px;
  width: 80px;
  padding: 15px 0;
  display: block;
  color: #fff;
  font-weight: bold;
  text-align: center;
  cursor: pointer;
}

.sign-out:hover,
.sign-out:focus {
  color: #999;
}

.brand-title {
  color: #fff;
  font-size: 14px;
  position: absolute;
  bottom: 100px;
  left: 28px;
  display: block;
  writing-mode: vertical-rl;
  -webkit-writing-mode: vertical-rl;
}

.brand-icon {
  display: block;
  width: 80px;
  height: 70px;
  text-align: center;
  color: #ccc;
  margin-top: 10px;
  background: no-repeat center center;
  background-size: 70%;
  text-indent: -9999px;
}

.white {
  text-transform: uppercase;
}

.framework-menu {
  width: 100%;
  height: 3em;
  background: #a50f15;
  position: fixed;
  top: 0;
  z-index: 1010;
  box-shadow: 0 1px 7px rgba(0, 0, 0, 0.4);

  .framework-brand {
    width: 80%;

    .hamburger-menu {
     display: none;
    }
  }
}

.framework-brand-title {
  color: #fff;
  position: absolute;
  display: block;
  top: 27%;
  font-size: 16px;
  left: 3em;
}

.framework-brand-icon {
  display: block;
  width: 32px;
  height: 32px;
  text-align: center;
  color: #ccc;
  margin-top: 5px;
  margin-left: 10px;
  background: no-repeat center center;
  background-size: 70%;
}

.framework-reports {
  right: 7%;
  top: 22%;
  position: absolute;

  #reports-divider {
    color: white;
    display: inline;
  }

  a {
    color: white;
    font-size: 10px;
    text-decoration: none;
  }

  a:hover {
    color: #333;
    text-decoration: none;
  }
}

.framework-sign-out {
  position: absolute;
  right: 0;
  font-size: 18px;
  width: 6%;
  padding: 9px 0;
  display: block;
  color: #fff;
  font-weight: bold;
  text-align: center;
  cursor: pointer;
  top: 0;
}

.framework-sign-out:hover,
.framework-sign-out:focus {
  color: #999;
}
