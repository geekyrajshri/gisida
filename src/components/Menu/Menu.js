require('./Menu.scss');

import React from 'react';
import Cookie from 'js-cookie';

const logOut = (e) => {
  e.preventDefault();
  Cookie.set('dsauth', false);
  location.reload();
}

const Menu = ({ toggleSingleScreen=f=>f, toggleSplitScreen=f=>f }) =>
<div className="menu">
  <div className="brand">
    <a className="brand-icon">UKAID</a>
    <div className="brand-title"><span className="white">Somalia</span>&nbsp;&nbsp; Integrated Response Platform</div>
  </div>
  <ul>
    <li><a href="#" onClick={toggleSingleScreen}><span className="glyphicon glyphicon-home"></span></a></li>
    <li><a href="#" onClick={toggleSplitScreen}><span className="glyphicon glyphicon-th-large"></span></a></li>
  </ul>
  <a className="sign-out" onClick={logOut}><span className="glyphicon glyphicon-log-out"></span></a>
</div>

export default Menu
