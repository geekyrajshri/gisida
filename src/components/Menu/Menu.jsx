import React from 'react';
import PropTypes from 'prop-types';
import Cookie from 'js-cookie';

require('./Menu.scss');

const logOut = (e) => {
  e.preventDefault();
  Cookie.set('dsauth', false);
  location.reload();
};

const Menu = ({ toggleSingleScreen = f => f, toggleSplitScreen = f => f, appConfig = f => f }) =>
  appConfig.defaultView === 'framework' ?
    (<div className="framework-menu">
      <div className="framework-brand">
        <img src={appConfig.appIcon} alt={appConfig.appName} className="framework-brand-icon" />
        <div className="framework-brand-title">
          <span className="framework-white">{appConfig.appName}</span>&nbsp;&nbsp; {appConfig.appNameDesc}</div>
      </div>
      <div className="framework-reports"><a>How do I use this Framework?</a> | <a>Download Reports</a></div>
      <a className="framework-sign-out" onClick={logOut} role="button" tabIndex={0}><span className="glyphicon glyphicon-log-out" /></a>
    </div>)
    :
    (<div className="menu-horizontal">
      <div className="brand">
        <img src={appConfig.appIcon} alt={appConfig.appName} className="brand-icon" />
        <div className="brand-title">
          <span className="white">{appConfig.appName}</span>&nbsp;&nbsp; {appConfig.appNameDesc}</div>
      </div>
      <ul>
        <li><a href="#" onClick={toggleSingleScreen}><span className="glyphicon glyphicon-home" /></a></li>
        <li><a href="#" onClick={toggleSplitScreen}><span className="glyphicon glyphicon-th-large" /></a></li>
      </ul>
      <a className="sign-out" onClick={logOut} role="button" tabIndex={0}><span className="glyphicon glyphicon-log-out" /></a>
    </div>);

Menu.propTypes = {
  toggleSingleScreen: PropTypes.func.isRequired,
  toggleSplitScreen: PropTypes.func.isRequired,
  appConfig: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default Menu;
