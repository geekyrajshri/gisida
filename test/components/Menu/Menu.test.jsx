import React from 'react';
import renderer from 'react-test-renderer';
import Menu from '../../../src/components/Menu/Menu';

describe('Menu', () => {
    it('Menu renders application details', () => {
        const component = renderer.create(<Menu />);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
});
