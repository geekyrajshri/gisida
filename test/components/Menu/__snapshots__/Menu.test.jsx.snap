// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Menu renders application details 1`] = `
<div
  className="menu"
>
  <div
    className="brand"
  >
    <img
      alt="UKAID"
      className="brand-icon"
      src={undefined}
    />
    <div
      className="brand-title"
    >
      <span
        className="white"
      />
         
    </div>
  </div>
  <ul>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        <span
          className="glyphicon glyphicon-home"
        />
      </a>
    </li>
    <li>
      <a
        href="#"
        onClick={[Function]}
      >
        <span
          className="glyphicon glyphicon-th-large"
        />
      </a>
    </li>
  </ul>
  <a
    className="sign-out"
    onClick={[Function]}
  >
    <span
      className="glyphicon glyphicon-log-out"
    />
  </a>
</div>
`;
